### DISCUSSION TITLE
Email-notifications on new comments

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#200] [Q&A] [answered]
Is there a way for Admins to receive notifications by default through email when somebody adds a comment to a blogpost without having to comment yourself and checking the "Send notification when somebody adds a new comment"?

I tested this by adding a comment (with Tor) by a fictional person and did not receive a notification of it.

If this is not possible for now I will try to make a new Module for it, or would it be better to (try to) modify the Comment-module like adding a checkbox "Get notifications on new comments"?

**Comment by xenocrat** [comment, marked as answer]
Hello there,

Currently this isn't possible, but it's a good idea for a feature. I will take a look at implementing it for the next release.

In the meantime you can indeed achieve the same thing with a module. It would look like this:

````
<?php
    class CommentMailer extends Modules {
        public function add_comment($comment) {
            $config = Config::current();
            $mailto = $config->email;

            $email_headers = "From: ".$config->email."\r\n"."X-Mailer: ".CHYRP_IDENTITY;
            $email_subject = _f("New Comment at %s", $config->name, "comments");
            $email_message = _f("%s commented on a blog post:", $comment->author, "comments").
                             "\r\n".
                             unfix($comment->url()).
                             "\r\n".
                             "\r\n".
                             truncate(strip_tags($comment->body), 60);

            email($mailto, $email_subject, $email_message, $email_headers);
        }
    }
````

This module responds to the "add_comment" trigger which is fired by the Comments module whenever a new comment is added to a post. I've named the module "CommentMailer" but you can name it whatever you like.

**Reply by eawmaas** [reply]
Ok, I might dive into it soon, when I can (hopefully) finish something else.

And thank you for the willingness to try to put it in the new release and the advise that followed, it helps a lot to get a strong hint in the right direction like this one!

**Comment by eawmaas** [comment]
Hi there!

I have made the Mod with Dutch translations added.
It was exactly your code, so in info.php all credits go to you.

I tested it (PHP 8.2) several times (not being logged in and with other IP-numbers through Tor) and it works like a breeze, very happy with it!
[commentmailer.zip](https://github.com/xenocrat/chyrp-lite/files/12657676/commentmailer.zip)
